lib/ansible/modules/cloud/amazon/ecs_taskdefinition_facts.py

```diff
@@ -0,0 +1,350 @@
+            returned: when present
+            type: string
+status:
+    description: The status of the task definition
+    returned: always
+    type: string
+requires_attributes:
+    description: The container instance attributes required by your task
+    returned: when present
+    type: complex
+    contains:
+        name:
+            description: The name of the attribute.
+            returned: when present
+            type: string
+        value:
+            description: The value of the attribute.
+            returned: when present
+            type: string
+        targetType:
+            description: The type of the target with which to attach the attribute.
+            returned: when present
+            type: string
+        targetId:
+            description: The ID of the target.
+            returned: when present
+            type: string
+placement_constraints:
+    description: A list of placement constraint objects to use for tasks
+    returned: always
+    type: complex
+    contains:
+        type:
+            description: The type of constraint.
+            returned: when present
+            type: string
+        expression:
+            description: A cluster query language expression to apply to the constraint.
+            returned: when present
+            type: string
+'''  # NOQA
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import _camel_to_snake, camel_dict_to_snake_dict, boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(dict(
+        task_definition=dict(required=True, type='str')
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=True)
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 is required.')
+
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        if not region:
+            module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
+        ecs = boto3_conn(module, conn_type='client', resource='ecs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.ProfileNotFound as e:
+        module.fail_json(msg=str(e))
+
+    ecs_td = ecs.describe_task_definition(taskDefinition=module.params['task_definition'])['taskDefinition']
+    ecs_td_snake = {}
+    for k, v in ecs_td.items():
+        ecs_td_snake[_camel_to_snake(k)] = v
+
+    # ecs_td = ecs.describe_task_definition(taskDefinition=module.params['task_definition'])['taskDefinition']
+    # ecs_td_container_definitions = ecs_td['containerDefinitions']
+    # del ecs_td['containerDefinitions']
```

Comment:
I think of the options I prefer the currently-implemented (not commented) version that changes the top-level, but leaves sub-options as they are, so IMO you can delete this commented stuff and it looks good to me.